### PR TITLE
Made the Rasterizer avoid GPU calls when backgrounded

### PR DESCRIPTION
**Pull request by gaaclarke**
relevant issue: https://github.com/flutter/flutter/issues/57464

### CODE REVIEW

File: shell/common/rasterizer.h [hunk 1]
```diff
@@ -103,7 +104,8 @@ class Rasterizer final : public SnapshotDelegate {
   ///                                 the GPU state used by the rasterizer.
   ///
   Rasterizer(TaskRunners task_runners,
-             std::unique_ptr<flutter::CompositorContext> compositor_context);
+             std::unique_ptr<flutter::CompositorContext> compositor_context,
+             std::shared_ptr<fml::SyncSwitch> is_gpu_disabled_sync_switch);
```
Comment (dnfield) [inline shell/common/rasterizer.h:108]: docs here and below for new param

File: shell/platform/fuchsia/flutter/engine.cc [hunk 2]
```diff
@@ -206,9 +206,9 @@ Engine::Engine(Delegate& delegate,
         }
 
         return std::make_unique<flutter::Rasterizer>(
-            shell.GetTaskRunners(),        // task runners
-            std::move(compositor_context)  // compositor context
-        );
+            /*task_runners=*/shell.GetTaskRunners(),
+            /*compositor_context=*/std::move(compositor_context),
+            /*is_gpu_disabled_sync_switch=*/shell.GetIsGpuDisabledSyncSwitch());
```
Comment (dnfield) [inline shell/platform/fuchsia/flutter/engine.cc:211]: nit: do we use this style anywhere else in the engine? Why change away from convention here?

File: shell/common/rasterizer.h (hunk 1 above)
Comment (gaaclarke) [reply shell/common/rasterizer.h:111]: done

File: shell/platform/fuchsia/flutter/engine.cc (hunk 2 above)
Comment (gaaclarke) [reply shell/platform/fuchsia/flutter/engine.cc:211]: Yes, this is a google standard, please look at google's other code to see millions of examples.  Their is even a linter that enforces that the names are correct, that is why it is standardized.

Comment (gaaclarke) [reply shell/platform/fuchsia/flutter/engine.cc:211]: https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments  Here is the rule in the c++ style guide.

**Pull request merged**